spec/factory_spec.rb

```diff
@@ -320,6 +320,32 @@ def to_msgpack_ext
         expect(factory.dump(42)).to eq(MessagePack.dump(42))
       end
     end
+
+    describe "registering ext type with manual serialization" do
+      before do
+        stub_const("Point", Struct.new(:x, :y, :z))
+      end
+
+      it 'can receive the packer as argument' do
+        factory = MessagePack::Factory.new
+        factory.register_type(0x00, Symbol)
+        factory.register_type(
+          0x01,
+          Point,
+          packer: ->(point, packer) do
+            packer.write(point.to_a)
+          end,
+          unpacker: ->(payload, unpacker) do
```

Comment:
I must admit I'm a bit annoyed by this `payload` param being useless.

Reply:
A solution might be to pass an option when registering the extension instead or relying on the `arity`. e.g.

```ruby
        factory.register_type(
          0x01,
          Point,
          packer: ->(point, packer) do
            packer.write(point.to_h)
            nil
          end,
          unpacker: ->(unpacker) do
            attrs = unpacker.read
            Point.new(attrs.fetch(:x), attrs.fetch(:y), attrs.fetch(:z))
          end,
          streaming: true,
        )
```

(doubt `streaming` is a good name but...)